R/theme.R

```diff
@@ -775,12 +774,12 @@ combine_elements <- function(e1, e2) {
 
   # Calculate relative sizes
   if (is.rel(e1$size)) {
-    e1$size <- e2$size * unclass(e1$size)
+    e1$size <- (e2$size %||% rel(1)) * unclass(e1$size)
   }
 
   # Calculate relative linewidth
   if (is.rel(e1$linewidth)) {
-    e1$linewidth <- e2$linewidth * unclass(e1$linewidth)
+    e1$linewidth <- (e2$linewidth %||% rel(1)) * unclass(e1$linewidth)
```

Comment:
If `e2` has `NULL` size/linewidth, this used to result in `numeric(0)` as size/linewidth. {grid} cannot deal with 0-length gpar settings.